src/BenchmarkDotNet/Analysers/RegressionAnalyser.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Mathematics.StatisticalTesting;
+using BenchmarkDotNet.Reports;
+
+namespace BenchmarkDotNet.Analysers
+{
+    public class RegressionAnalyser : AnalyserBase
+    {
+        public static readonly IAnalyser Default = new RegressionAnalyser(RelativeThreshold.Default);
+        
+        public RegressionAnalyser(Threshold threshold) => Threshold = threshold;
+
+        public override string Id => "Regression";
+
+        private Threshold Threshold { get; }
+
+        protected override IEnumerable<Conclusion> AnalyseSummary(Summary summary)
+        {
+            foreach (var sameArguments in summary.BenchmarksCases.GroupBy(benchmark => (benchmark.Descriptor, benchmark.Parameters)))
+            {
+                var benchmarks = sameArguments.ToArray();
+                if (benchmarks.Length != 2)
```

Comment:
Do we really need this limitation? What if I want to compare the baseline against several other jobs?